src/wp-includes/blocks.php

```diff
@@ -326,13 +326,15 @@ function register_block_type_from_metadata( $file_or_folder, $args = array() ) {
 		trailingslashit( $file_or_folder ) . 'block.json' :
 		$file_or_folder;
 
-	if ( ! file_exists( $metadata_file ) ) {
+	$is_core_block = str_starts_with( $file_or_folder, ABSPATH . WPINC );
+
+	if ( ! $is_core_block && ! file_exists( $metadata_file ) ) {
```

Comment:
I'm wondering if we can remove the `file_exists` check entirely from this function? Below, this function calls `wp_json_file_decode()` on the file path if it can't be found in the cache for core meta. That function will return `null` if the file doesn't exists, which would also cause this function to return `false`. Doing an early `file_exists` check might be a bit faster when the file does not actually exist, but most of the time, I expect this is being called for files that do exists and error handling down the line will take care of things just fine.